### PR TITLE
mgr/dashboard: save mgr logs inside a folder 

### PR DESCRIPTION
**Pull request by nizamial09**
Save the mgr container logs of cephadm inside a folder and later on
archive it and get it as an artifact on the cephadm dashboard e2e jobs

this saves the mgr logs in `ceph/logs/mgr.cephadm.log`
and also the container logs and journal logs in the same folder.

Fixes: https://tracker.ceph.com/issues/55247
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Review by pereman2 — APPROVED** [review]

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh [hunk 1]
```diff
@@ -18,16 +18,16 @@ on_error() {
     if [ "$1" != "0" ]; then
         printf "\n\nERROR $1 thrown on line $2\n\n"
         printf "\n\nCollecting info...\n\n"
-        printf "\n\nDisplaying MGR logs:\n\n"
-        kcli ssh -u root -- ceph-node-00 'cephadm logs -n \$(cephadm ls | grep -Eo "mgr\.ceph[0-9a-z.-]+" | head -n 1) -- --no-tail --no-pager'
-        for vm_id in 0 1 2
+        printf "\nSaving MGR logs:\n"
+        mkdir -p ${CEPH_DEV_FOLDER}/logs
+        kcli ssh -u root -- ceph-node-00 'cephadm logs -n \$(cephadm ls | grep -Eo "mgr\.ceph[0-9a-z.-]+" | head -n 1) -- --no-tail --no-pager' > ${CEPH_DEV_FOLDER}/logs/mgr.cephadm.log
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh:23]: Aren't these logs somehow sent to `journald`? Can't they be filtered with a `journalctl` command? Perhaps something like `journalctl -n all -ft cloud-init` would simplify a lot this expression.

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh [hunk 2]
```diff
@@ -18,16 +18,16 @@ on_error() {
     if [ "$1" != "0" ]; then
         printf "\n\nERROR $1 thrown on line $2\n\n"
         printf "\n\nCollecting info...\n\n"
-        printf "\n\nDisplaying MGR logs:\n\n"
-        kcli ssh -u root -- ceph-node-00 'cephadm logs -n \$(cephadm ls | grep -Eo "mgr\.ceph[0-9a-z.-]+" | head -n 1) -- --no-tail --no-pager'
-        for vm_id in 0 1 2
+        printf "\nSaving MGR logs:\n"
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh:21]: `printf` is ok but for simple interpolation I'd use `echo`
```sh
        echo "ERROR $1 thrown on line $2"
        echo
        echo "Collecting info..."
        echo
        echo "Saving MGR logs:"
```

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh [hunk 3]
```diff
@@ -18,16 +18,16 @@ on_error() {
...
+        mkdir -p ${CEPH_DEV_FOLDER}/logs
+        kcli ssh -u root -- ceph-node-00 'cephadm logs -n \$(cephadm ls | grep -Eo "mgr\.ceph[0-9a-z.-]+" | head -n 1) -- --no-tail --no-pager' > ${CEPH_DEV_FOLDER}/logs/mgr.cephadm.log
+        for vm_id in 0 1 2 3
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh:24]:
```suggestion
        for vm_id in {0..3}
```

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh [hunk 4]
```diff
@@ -18,16 +18,16 @@ on_error() {
...
         do
             local vm="ceph-node-0${vm_id}"
-            printf "\n\nDisplaying journalctl from VM ${vm}:\n\n"
-            kcli ssh -u root -- ${vm} 'journalctl --no-tail --no-pager -t cloud-init' || true
-            printf "\n\nEnd of journalctl from VM ${vm}\n\n"
-            printf "\n\nDisplaying container logs:\n\n"
-            kcli ssh -u root -- ${vm} 'podman logs --names --since 30s \$(podman ps -aq)' || true
+            printf "\nSaving journalctl from VM ${vm}:\n"
+            kcli ssh -u root -- ${vm} 'journalctl --no-tail --no-pager -t cloud-init' > ${CEPH_DEV_FOLDER}/logs/journal.ceph-node-0${vm_id}.log || true
+            printf "\nSaving container logs:\n"
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh:29]: Same here: let's replace `printf` with `echo`.

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh:23]: `journalctl -n all -ft cloud-init` displays the cephadm bootstrap logs of the ceph-node-00. What we want over here is the mgr container logs and its fetched using `cephadm logs` command.

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh (hunk 2 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh]: done!

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh (hunk 3 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh]: done!

File: src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh (hunk 4 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh]: done!

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test dashboard

**Review by epuertat — APPROVED** [review]

**Pull request merged**